depends/hosts/darwin.mk

```diff
@@ -1,17 +1,50 @@
-OSX_MIN_VERSION=10.12
-OSX_SDK_VERSION=10.12
-OSX_SDK=$(SDK_PATH)/MacOSX$(OSX_SDK_VERSION).sdk
-LD64_VERSION=253.9
-darwin_CC=gcc-8 -target $(host) -mmacosx-version-min=$(OSX_MIN_VERSION) --sysroot $(OSX_SDK) -mlinker-version=$(LD64_VERSION)
-darwin_CXX=g++-8 -target $(host) -mmacosx-version-min=$(OSX_MIN_VERSION) --sysroot $(OSX_SDK) -mlinker-version=$(LD64_VERSION)
+OSX_MIN_VERSION=10.15
+OSX_SDK_VERSION=10.15.6
+XCODE_VERSION=12.1
+XCODE_BUILD_ID=12A7403
+LD64_VERSION=609
+
+OSX_SDK=$(SDK_PATH)/Xcode-$(XCODE_VERSION)-$(XCODE_BUILD_ID)-extracted-SDK-with-libcxx-headers
+
+# Flag explanations:
+#
+#     -mlinker-version
+#
+#         Ensures that modern linker features are enabled. See here for more
+#         details: https://github.com/bitcoin/bitcoin/pull/19407.
+#
+#     -B$(build_prefix)/bin
+#
+#         Explicitly point to our binaries (e.g. cctools) so that they are
+#         ensured to be found and preferred over other possibilities.
+#
+#     -nostdinc++ -isystem $(OSX_SDK)/usr/include/c++/v1
+#
+#         Forces clang to use the libc++ headers from our SDK and completely
+#         forget about the libc++ headers from the standard directories
+#
+#         TODO: Once we start requiring a clang version that has the
+#         -stdlib++-isystem<directory> flag first introduced here:
+#         https://reviews.llvm.org/D64089, we should use that instead. Read the
+#         differential summary there for more details.
+#
+darwin_CC=clang -target $(host) -mmacosx-version-min=$(OSX_MIN_VERSION) --sysroot $(OSX_SDK) -mlinker-version=$(LD64_VERSION) -B$(build_prefix)/bin
+darwin_CXX=clang++ -target $(host) -mmacosx-version-min=$(OSX_MIN_VERSION) --sysroot $(OSX_SDK) -stdlib=libc++ -mlinker-version=$(LD64_VERSION) -B$(build_prefix)/bin -nostdinc++ -isystem $(OSX_SDK)/usr/include/c++/v1
 
 darwin_CFLAGS=-pipe
 darwin_CXXFLAGS=$(darwin_CFLAGS)
 
-darwin_release_CFLAGS=-g -O2
+darwin_release_CFLAGS=-O2
 darwin_release_CXXFLAGS=$(darwin_release_CFLAGS)
 
-darwin_debug_CFLAGS=-g -O0
+darwin_debug_CFLAGS=-O1
```

Comment:
it was important (for gcc at least) to have -O0 and -g flags in debug build to disable all optimisations and be able to watch all variables under debug. 
I am worried that this -O1 flag may hide vars under debug

Reply:
Instead of using `DEBUG=1`, you can always pass `CXXFLAGS='-g -O0'` to `./configure`. However, we will later set the old `darwin_debug_CFLAGS`.

Currently, the issue is quite peculiar. On my Mac host, everything builds perfectly, resulting in a `komodod` that is a `Mach-O 64-bit executable x86_64`. However, on the GitHub hosted runner, we are getting an invalid `komodod` binary (you can view it in the artifacts). I have not yet been able to determine why this is happening.

Reply:
btw I saw a similar issue on CI with komodo mac failed to run with OSError: [Errno 8] Exec format error. This started to happen 2 years ago

Reply:
I can provide more information; it depends on the versions of MacOS, XCode, and Rust being used. A poor combination of these three could result in a non-executable file with a series of zeros at the beginning. However, the Rust version is up to date and the MacOS and XCode versions on the runner appear to be good. I am still investigating to find the cause.

Reply:
![image](https://github.com/KomodoPlatform/komodo/assets/22120003/79b848c0-0135-4830-8c60-979432a61cb3)

Using other builder host - result is also correct, but on runner host we still getting invalid binary. Mystery.